lib/elixir/test/elixir/kernel/expansion_test.exs

```diff
@@ -1988,6 +1988,31 @@ defmodule Kernel.ExpansionTest do
       assert expand(quote(do: <<"foo", <<x::size(4), y::size(4)>>::binary>>))
              |> clean_meta([:alignment]) ==
                quote(do: <<"foo"::binary(), x()::integer()-size(4), y()::integer()-size(4)>>)
+
+      before_expansion =
+        quote do
+          b = "b"
+          "a" <> ^b = "ab"
+        end
+
+      after_expansion =
+        quote do
+          b = "b"
+          <<"a"::binary(), ^b::binary()>> = "ab"
```

Comment:
I have the impression that if the expansion was
```elixir
<<"a"::binary(), ^b>> = "ab"
```

Would be possible to allow the operator on the left side. In Erlang:
```erlang
23> A. 
97
24> <<A, "b">> = <<"ab">>.       
<<"ab">>
25> <<A/binary, "b">> = <<"ab">>.
* 1: a binary field without size is only allowed at the end of a binary pattern
```